### PR TITLE
removes excessive runs from bats tests

### PR DESCRIPTION
**Pull request by wgahnagl**
#### What type of PR is this?

> /kind api-change
> /kind bug

/kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:
Removes excessive runs from the bats tests
#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Comment by kolyshkin** [comment]
There are some indentation issues (as seen in https://app.circleci.com/pipelines/github/cri-o/cri-o/6334/workflows/e5fd94c2-903f-40b5-aa80-70402561c757/jobs/101963, or you can just run `make shfmt`).

The `integration-cgroups` failure in `pipeline` CI job appears to be a flake (caused by slow circleci?)

File: test/crio-wipe.bats [hunk 1]
```diff
@@ -32,35 +32,24 @@ function run_crio_wipe() {
...
 function test_crio_wiped_containers() {
-	run crictl pods -v
-	[ "$status" -eq 0 ]
-	[ "$output" == "" ]
-
-	run crictl ps -v
-	[ "$status" -eq 0 ]
-	[ "$output" == "" ]
+	crictl pods -v
+	crictl ps -v
```
Comment (kolyshkin) [inline test/crio-wipe.bats:36]: So, this removes check that the output from these two commands is empty.

This should be
```sh
output=$(crictl pods -v)
[ "$output" = "" ]
# (same for ps)
```

File: test/crio-wipe.bats [hunk 2]
```diff
@@ -32,35 +32,24 @@ function run_crio_wipe() {
...
 }
 
 function test_crio_did_not_wipe_containers() {
-	run crictl pods -v
-	[ "$status" -eq 0 ]
-	[ "$output" != "" ]
+	crictl pods -v
```
Comment (kolyshkin) [inline test/crio-wipe.bats:40]: similar issue here -- there was a check that `crictl pods -v` actually printed something (i.e. output is non-empty), which this commit removes.

File: test/ctr_seccomp.bats [hunk 3]
```diff
@@ -23,39 +23,20 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
-	run crictl create "$pod_id" "$TESTDIR"/seccomp1.json "$TESTDATA"/sandbox_config.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	ctr_id="$output"
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	ctr_id=$(crictl create "$pod_id" "$TESTDIR"/seccomp1.json "$TESTDATA"/sandbox_config.json)
 	run crictl start "$ctr_id"
```
Comment (kolyshkin) [inline test/ctr_seccomp.bats:28]: whoops, forgot to remove `run `. This way the exit code is never checked.

File: test/ctr_seccomp.bats [hunk 4]
```diff
@@ -23,39 +23,20 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
 	run crictl exec --sync "$ctr_id" chmod 777 .
```
Comment (kolyshkin) [inline test/ctr_seccomp.bats:29]: ditto

File: test/ctr_seccomp.bats [hunk 5]
```diff
@@ -64,31 +45,17 @@ function teardown() {
...
 
 # 4. test running with ctr wrong profile name
 @test "ctr seccomp profiles wrong profile name" {
 	sed -e 's/%VALUE%/wontwork/g' "$TESTDATA"/container_config_seccomp.json > "$TESTDIR"/seccomp1.json
-	run crictl runp "$TESTDATA"/sandbox_config.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
 	run crictl create "$pod_id" "$TESTDIR"/seccomp1.json "$TESTDATA"/sandbox_config.json
```
Comment (kolyshkin) [inline test/ctr_seccomp.bats:58]: ditto

File: test/image.bats [hunk 6]
```diff
@@ -22,193 +22,117 @@ function teardown() {
...
 @test "container status when created by image ID" {
 	start_crio
 
-	run crictl runp "$TESTDATA"/sandbox_config.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	
+    sed -e "s/%VALUE%/$REDIS_IMAGEID/g" "$TESTDATA"/container_config_by_imageid.json > "$TESTDIR"/ctr_by_imageid.json
```
Comment (kolyshkin) [inline test/image.bats:36]: indentation issue?

File: test/image.bats [hunk 7]
```diff
@@ -22,193 +22,117 @@ function teardown() {
...
 
-	run crictl runp "$TESTDATA"/sandbox_config.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
-
-	sed -e "s/%VALUE%/quay.io\/crio\/redis:alpine/g" "$TESTDATA"/container_config_by_imageid.json > "$TESTDIR"/ctr_by_imagetag.json
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	
+    sed -e "s/%VALUE%/quay.io\/crio\/redis:alpine/g" "$TESTDATA"/container_config_by_imageid.json > "$TESTDIR"/ctr_by_imagetag.json
```
Comment (kolyshkin) [inline test/image.bats:50]: ditto

File: test/image.bats [hunk 8]
```diff
@@ -22,193 +22,117 @@ function teardown() {
...
 
-	run crictl runp "$TESTDATA"/sandbox_config.json
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_id="$output"
-
-	sed -e "s|%VALUE%|$IMAGE_LIST_DIGEST|g" -e 's|"/bin/ls"|"/bin/sleep", "1d"|g' "$TESTDATA"/container_config_by_imageid.json > "$TESTDIR"/ctr_by_imagelistref.json
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	
+    sed -e "s|%VALUE%|$IMAGE_LIST_DIGEST|g" -e 's|"/bin/ls"|"/bin/sleep", "1d"|g' "$TESTDATA"/container_config_by_imageid.json > "$TESTDIR"/ctr_by_imagelistref.json
```
Comment (kolyshkin) [inline test/image.bats:79]: ditto

File: test/image.bats [hunk 9]
```diff
@@ -22,193 +22,117 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
+	crictl pull quay.io/crio/nginx@sha256:1ad874092a55efe2be0507a01d8a300e286f8137510854606ab1dd28861507a3
 
-	run crictl images --quiet quay.io/crio/nginx@sha256:1ad874092a55efe2be0507a01d8a300e286f8137510854606ab1dd28861507a3
-	[ "$status" -eq 0 ]
-	echo "$output"
-	[ "$output" != "" ]
-	imageid="$output"
+	imageid=$(crictl images --quiet quay.io/crio/nginx@sha256:1ad874092a55efe2be0507a01d8a300e286f8137510854606ab1dd28861507a3)
```
Comment (kolyshkin) [inline test/image.bats:130]: missing check for non-empty $imageid

File: test/image.bats [hunk 10]
```diff
@@ -465,67 +316,41 @@ function teardown() {
...
+		crictl rmi "$id"
 	done
-	run crictl images --quiet
-	echo "$output"
-	[ "$status" -eq 0 ]
-
-	cleanup_images
+	crictl images --quiet
+	
+    cleanup_images
```
Comment (kolyshkin) [inline test/image.bats:326]: indentation issue

File: test/inspecti.bats [hunk 11]
```diff
@@ -13,6 +13,6 @@ function teardown() {
 }
 
 @test "inspect image should succed contain all necessary information" {
-	run crictl inspecti quay.io/crio/redis:alpine | jq -e .status.size
-	run crictl inspecti quay.io/crio/redis:alpine | jq -e .info.imageSpec.config.Cmd
```
Comment (kolyshkin) [inline test/inspecti.bats:17]: The funny thing is, this was not working at all because
1. `run` ate all the output.
2. `jq` just ignore empty input.

Now it's working, but the issue 2 still remains, so the correct way to fix this is
```sh
output=$(crictl inspecti quay.io/crio/redis:alpine)
[ "$output" != "" ]
jq -e '.status.size'<<< "$output"
jq -e '.info.imageSpec.config.Cmd' <<< "$output"
```

(can and should be addressed by a separate commit; same PR is fine)

File: test/restore.bats [hunk 12]
```diff
@@ -12,90 +12,51 @@ function teardown() {
...
-	pod_list_info="$output"
-
-	run crictl inspectp -o json "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	pod_list_info=$(crictl pods --quiet --id "$pod_id")
+
+	output=$(crictl inspectp -o json "$pod_id")
 	pod_status_info=$(echo "$output" | jq ".status.state")
```
Comment (kolyshkin) [inline test/restore.bats:19]: Since `jq` ignores empty input (see previous comment), there should be a check for non-empty output added here. Can be addressed either right here or separately.

File: test/restore.bats [hunk 13]
```diff
@@ -12,90 +12,51 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
+	output=$(crictl pods --quiet --id "$pod_id")
 	[[ "${output}" == "${pod_list_info}" ]]
 
-	run crictl inspectp -o json "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
+	output=$(crictl inspectp -o json "$pod_id")
 	status_output=$(echo "$output" | jq ".status.state")
```
Comment (kolyshkin) [inline test/restore.bats:38]: check for empty output needed because of `jq` (see above comments)

File: test/restore.bats [hunk 14]
```diff
@@ -12,90 +12,51 @@ function teardown() {
...
 	created_at_output=$(echo "$output" | jq ".status.createdAt")
 	[[ "${status_output}" == "${pod_status_info}" ]]
 	[[ "${ip_output}" == "${pod_ip}" ]]
 	[[ "${created_at_output}" == "${pod_created_at}" ]]
 
-	run crictl ps --quiet --all
-	echo "$output"
-	[ "$status" -eq 0 ]
-	[[ "${output}" != "" ]]
+	output=$(crictl ps --quiet --all)
```
Comment (kolyshkin) [inline test/restore.bats:45]: check for non-empty output is removed but that's OK because it was redundance since we check for a particular string below.

File: test/restore.bats (hunk 13 above)
Comment (kolyshkin) [reply test/restore.bats:38]: and `jq` probably needs `-e` so it errors out if `.status.state` is not found

File: test/restore.bats [hunk 15]
```diff
@@ -229,50 +144,29 @@ function teardown() {
...
-	echo "$output"
-	[ "$status" -eq 0 ]
+	output=$(crictl inspectp "$pod_id")
 	[[ "${output}" == *"SANDBOX_NOTREADY"* ]]
 
-	run crictl ps --quiet --all
-	echo "$output"
-	[ "$status" -eq 0 ]
-	[[ "${output}" != "" ]]
+	output=$(crictl ps --quiet --all)
```
Comment (kolyshkin) [inline test/restore.bats:153]: non-empty output check is removed but it's OK since we have a check for a concrete value below.

File: test/restore.bats [hunk 16]
```diff
@@ -283,243 +177,133 @@ function teardown() {
...
-	[ "$status" -eq 0 ]
-	ctr_id="$output"
-
-	run crictl ps --quiet --id "$ctr_id" --all
-	echo "$output"
-	[ "$status" -eq 0 ]
-	ctr_list_info="$output"
+	ctr_id=$(crictl create "$pod_id" "$TESTDATA"/container_config.json "$TESTDATA"/sandbox_config.json)
+	
+    ctr_list_info=$(crictl ps --quiet --id "$ctr_id" --all)
```
Comment (kolyshkin) [inline test/restore.bats:200]: indentation issue

File: test/restore.bats [hunk 17]
```diff
@@ -283,243 +177,133 @@ function teardown() {
...
-	[ "$status" -eq 0 ]
-	pod_list_info="$output"
-
-	run crictl inspectp -o table "$pod_id"
-	echo "$output"
-	[ "$status" -eq 0 ]
-	pod_status_info=$(echo "$output" | grep ^Status)
+	pod_id=$(crictl runp "$TESTDATA"/sandbox_config.json)
+	pod_list_info=$(crictl pods --quiet --id "$pod_id"
+    )
```
Comment (kolyshkin) [inline test/restore.bats:235]: whoops, this should be on the previous line

File: test/restore.bats [hunk 18]
```diff
@@ -283,243 +177,133 @@ function teardown() {
...
+	output=$(crictl inspectp -o table "$pod_id")	
+    pod_status_info=$(echo "$output" | grep ^Status)
```
Comment (kolyshkin) [inline test/restore.bats:237]: indentation issue

**Review by kolyshkin — APPROVED** [review]
Good job overall, left a few issues that need to be addressed.

**Review by kolyshkin — CHANGES_REQUESTED** [review]
left a few suggestions

File: test/ctr_seccomp.bats (hunk 4 above)
Comment (wgahnagl) [reply test/ctr_seccomp.bats:29]: I feel like here I actually should be using run, because it's checking the status and the output ?

File: test/crio-wipe.bats (hunk 1 above)
Comment (wgahnagl) [reply test/crio-wipe.bats]: fixed!

File: test/ctr_seccomp.bats (hunk 3 above)
Comment (wgahnagl) [reply test/ctr_seccomp.bats]: fixed!

File: test/image.bats (hunk 6 above)
Comment (wgahnagl) [reply test/image.bats]: fixed!

File: test/image.bats (hunk 8 above)
Comment (wgahnagl) [reply test/image.bats]: fixed!

File: test/image.bats (hunk 9 above)
Comment (wgahnagl) [reply test/image.bats:130]: fixed!

File: test/image.bats (hunk 10 above)
Comment (wgahnagl) [reply test/image.bats]: fixed!

File: test/restore.bats (hunk 16 above)
Comment (wgahnagl) [reply test/restore.bats]: fixed!

File: test/restore.bats (hunk 18 above)
Comment (wgahnagl) [reply test/restore.bats]: fixed!

File: test/inspecti.bats (hunk 11 above)
Comment (kolyshkin) [reply test/inspecti.bats:17]: Just tested on my machine

```console
[kir@kir-rhat selinux]$ out=$(sudo crictl inspecti quay.io/crio/redis:alpine)
[kir@kir-rhat selinux]$ jq -e '.status.size' <<<$out
"28148166"
```

File: test/crio-wipe.bats (hunk 2 above)
Comment (wgahnagl) [reply test/crio-wipe.bats]: fixed!

**Comment by kolyshkin** [comment]
One single shfmt issue remaining; otherwise I think one this is good to go
https://app.circleci.com/pipelines/github/cri-o/cri-o/6433/workflows/480e8334-0c16-4791-a740-ff02d0b2b8e4/jobs/103551

**Review by kolyshkin — APPROVED** [review]
LGTM

**Review by mrunalp — APPROVED** [review]
